torch/ao/quantization/quantizer/x86_inductor_quantizer.py

```diff
@@ -270,9 +284,32 @@ def set_global(self, quantization_config: QuantizationConfig):
     def set_config_for_operator_type(
         self, operator_type: str, quantization_config: QuantizationConfig
     ):
-        self.operator_type_config[operator_type] = quantization_config
+        warnings.warn(
+            "The function set_config_for_operator_type has been deprecated. Please use set_operator_type instead."
+        )
+
+    def set_operator_type(
```

Comment:
I think it's probably better to stay in the user facing things like `nn.Linear`, 'nn.functional.linear` instead of aten ops, since modeling user are not interacting with aten ops

Reply:
Thanks for the comment and agree. How about we:

- Change `set_operator_type` as a internal interface as `_set_aten_operator_type` and `_get_aten_operator_type` for implementation.
- Add new user-face API like `set_module_type` and `set_function_type` to accept parameters of `nn.Linear` as module and `nn.funcational.linear` or `torch.matmul` as function.

Reply:
this sounds good, please update summary as well.

please add the user-facing API as well, although this is starting to look like the old qconfig mapping API, but you get to choose what API to expose for users this time

Reply:
Thanks for the suggestion @jerryzh168. Please kindly help to take a look again.

- Introducing user-facing API as `set_function_type_qconfig` and `set_module_type_qconfig`.
- Summary has been updated.